src/pal/src/map/virtual.cpp

```diff
@@ -826,10 +832,8 @@ static LPVOID VIRTUALResetMemory(
 
     TRACE( "Resetting the memory now..\n");
 
-    StartBoundary = (UINT_PTR)lpAddress & ~VIRTUAL_PAGE_MASK;
-    // Add the sizes, and round down to the nearest page boundary. 
-    MemSize = ( ((UINT_PTR)lpAddress + dwSize + VIRTUAL_PAGE_MASK) & ~VIRTUAL_PAGE_MASK ) - 
-               StartBoundary;
+    StartBoundary = (UINT_PTR) ALIGN_DOWN(lpAddress, GetVirtualPageSize());
+    MemSize = ALIGN_UP((UINT_PTR)lpAddress + dwSize, GetVirtualPageSize()) - StartBoundary;
```

Comment:
I am not sure I understand why would it be functionally different. 
The `ALIGN_UP` does `result = (val + (alignment - 1)) & ~(alignment - 1);` 
The `ALIGN_DOWN` does `result = val & ~(alignment - 1);`

So `ALIGN_DOWN(lpAddress, GetVirtualPageSize())` gets `lpAddress & ~(GetVirtualPageSize() - 1)`
And `ALIGN_UP((UINT_PTR)lpAddress + dwSize, GetVirtualPageSize())` gets `(lpAddress + dwSize + (GetVirtualPageSize() - 1)) & ~(GetVirtualPageSize() - 1)`

Using the previous state and replacing VIRTUAL_PAGE_MASK by (GetVirtualPageSize() - 1) yields the same expressions.